lib/media.js

```diff
@@ -57,42 +52,100 @@ function findTop(str) {
  * @returns {string}
  */
 function splice(str, index, items) {
-  return str.substr(0, index) + _.flattenDeep(items).join('\n') + str.substr(index);
+  return str.substr(0, index) + items + str.substr(index);
+}
+
+/**
+ * Get the contents of a string that come after specified characters.
+ *
+ * @param  {string} str Any string to split
+ * @param  {string} dir The directory the file is in
+ * @return {string}
+ */
+function getFileName(str, dir) {
+  return str.split(dir)[1];
+}
+
+/**
+ * Retrieve the contents of a file
+ *
+ * @param  {array} fileArray   An array of file names
+ * @param  {string} targetDir  The directory to retrieve files from
+ * @param  {string} filePrefix The string that comes right before the file name
+ * @return {Promise}
+ */
+function getContentsOfFiles(fileArray, targetDir, filePrefix) {
+  var allPromises = [],
+    rootDir = process.cwd();
+
+  fileArray.forEach(file => {
+    allPromises.push(files.readFilePromise(path.join(rootDir, targetDir, getFileName(file, filePrefix))));
+  });
+
+  return Promise.all(allPromises)
+    .catch(err => {
+      throw err;
```

Comment:
Hmm the only step that's higher in the stack is the actual reading of the file's contents, and I didn't want to throw it there because each module that uses the function should be able to determine how the error is handled.